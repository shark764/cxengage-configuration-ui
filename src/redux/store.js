/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';
import { createEpicMiddleware } from 'redux-observable';
import { rootEpic } from './epics';
import reducer from './reducers';

const composeEnhancers = composeWithDevTools({
  name: window.location.hash === '#/supervisorToolbar' ? 'Cx Supervisor Toolbar' : 'Cx Configuration'
});
export default createStore(reducer, composeEnhancers(applyMiddleware(createEpicMiddleware(rootEpic))));
