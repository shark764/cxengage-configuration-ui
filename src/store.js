import { createStore } from 'redux';
import { fromJS } from 'immutable';
import reducer from './reducers';

export default function configureStore(initialState = {}) {
  return createStore(reducer, fromJS(initialState));
}
