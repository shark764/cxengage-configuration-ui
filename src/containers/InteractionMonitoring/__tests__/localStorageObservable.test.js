import { from } from 'rxjs/observable/from';
import { fromEvent } from 'rxjs/observable/fromEvent';
import {
  localStorageSubscribe,
  localStorageUnsubscribe
} from '../localStorageObservable';
import store from '../../../redux/store.js';

jest.mock('rxjs/observable/fromEvent');
jest.mock('../../../redux/store.js');

describe('Observable Test chaneges in localStorage to dispatched actions', () => {
  it(`updates the state when localstorage is changed`, done => {
    const messageEvent = from([{ status: 'local storage updated' }]);
    fromEvent.mockImplementationOnce(() => messageEvent);
    store.dispatch.mockClear();
    localStorageSubscribe();
    localStorageUnsubscribe();
    expect(store.dispatch).toBeCalledWith(
      expect.objectContaining({
        type: expect.any(String)
      })
    );
    expect(store.dispatch).toMatchSnapshot();
    done();
  });
});
