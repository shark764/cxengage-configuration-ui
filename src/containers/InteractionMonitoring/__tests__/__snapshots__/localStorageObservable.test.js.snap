// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Observable Test chaneges in localStorage to dispatched actions updates the state when localstorage is changed 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "state": null,
        "type": "UPDATE_SUPERVISOR_TOOLBAR_STATE",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
